### ISSUE TITLE
I don't want the internet permission

### ISSUE THREAD

**Issue by serrq** [DanielRendox/GroceryGenius#12]
Might you adding the groceries list package in Github download section?

Or including it directly in the apk.

**Comment by vdbhb59** [comment]
The list is available here: assets/product
Though I agree it seems to be calling it.
assets/product/default_products_en.json
assets/product/default_products_change_list.json

**Comment by DanielRendox** [comment]
@serrq Hmm, interesting.

Including it in the apk is not an option since I'd want to add real time grocery list sharing in the future. 

It currently downloads the list from GitHub on first install. But it is also set up to update the local version once any additional groceries or icons are added. This way, there is no need to ship a new version just to add some more groceries. 

However, without an internet permission users will need to do all of that manually. 🤔

**Comment by serrq** [comment]
> without an internet permission users will need to do all of that manually.

Just add an URL in app to link to download file in Github. Push it and it open an external browser page. That's it.

**Comment by vdbhb59** [comment]
Or better, allow upload via a file. That way there is no need for the app itself to connect to internet. Just upload and set in tables.

**Comment by serrq** [comment]
I already downloaded the file through Github.

Wait for upload via file feature in next versions. Or I change app target.

**Comment by serrq** [comment]
Currently the file is something.txt. We can renaming it in .bin once downloaded if needed.

**Comment by serrq** [comment]
@vdbhb59 I mean a copyable URL (in app) in order to suggest to non-tech savvy people where is this damned file.

An user can paste it where it wants (its browser).

**Comment by vdbhb59** [comment]
"Damned file" lol, nice one mate.

No I knew what you meant. I offered an alternate solution, as trust me people unlike tech savvy would love to download files (I can show you same file downloaded several times, lols, and they still download it over and over again. But yes, link sounds much better.
On more suggestion is, provide the option to download/use the file. Honestly, a lot of those items in the list are generic, or only for European end of items + non-veg is included as well.
The way to get things can increase the code base as well. The best approach is either via a link or upload functionality. Link probably wold require curl/wget logic, manual upload would increase the code base as well.

Just my thoughts. :)

**Comment by vdbhb59** [comment]
Another thought is: Can this connection to internet be a one time activity, post which the permission is automatically revoked, or can be revoked by user in some manner?

**Comment by serrq** [comment]
There is no revoke toggle for internet permission, currently. In Android, I mean.

**Comment by vdbhb59** [comment]
> There is no revoke toggle for internet permission, currently. In Android, I mean.

Yeah, I thought so. Was not sure, so thought of putting it here as an ask as well.

**Comment by Helveg** [comment]
What about bundling the file as a fallback for those that reject the internet permission? Unless it bloats the bundle, then you could consider a Dynamic Feature Module but that seems like more work? You switch to an Android App Bundle and keep an as-up-to-date as possible fallback dataset bundled in the feature module. When a user rejects the internet permission you install the dynamic module as a fallback mechanism.

**Comment by vdbhb59** [comment]
> What about bundling the file as a fallback for those that reject the internet permission? Unless it bloats the bundle, then you could consider a Dynamic Feature Module but that seems like more work? You switch to an Android App Bundle and keep an as-up-to-date as possible fallback dataset bundled in the feature module. When a user rejects the internet permission you install the dynamic module as a fallback mechanism.

When you say app bundle, as in split-apks? Then F-Droid will have issues, as it does not support split-apks. Developer will have more work to do as well. Or do you mean as an OBB? That maybe possible, I believe, but again too much of a build time, heavy output, etc.

**Comment by Helveg** [comment]
Ok, didn't know F-Droid doesn't support them
